tests/TurboMqtt.Tests/Protocol/Mqtt311DecoderSpecs.cs

```diff
@@ -197,5 +204,35 @@ public void ShouldHandleMultipleMessagesWithPartialFrame()
             decodedPackets3[0].Should().BeEquivalentTo(pingRespPacket);
             decodedPackets3[1].Should().BeEquivalentTo(publishPacket3, options => options.Excluding(x => x.Payload));
         }
+
+        [FsCheck.Xunit.Property(MaxTest = 1000)]
+        public Property TestPacketReassembly(MqttPacket packet)
+        {
+            var decoder = new Mqtt311Decoder();
+
+            var fragmentedPackets = PacketGenerators.FragmentedPackets(Arb.From(Gen.Constant(packet))).Generator
+                .Sample(0, 1).First();
+            
+            var estimatedSize = MqttPacketSizeEstimator.EstimateMqtt3PacketSize(packet);
+            var fullSize = MqttPacketSizeEstimator.GetPacketLengthHeaderSize(estimatedSize) + estimatedSize + 1;
```

Comment:
Added a +1 here. `GetPacketLengthHeaderSize` only calculates the payload length, still need to add 1 for the control byte

Reply:
LGTM